src/Raven.Server/EventListener/ContentionEventsHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.EventListener;
+
+public class ContentionEventsHandler : AbstractEventsHandler<ContentionEventsHandler.ContentionEvent>
+{
+    protected override HashSet<EventType> DefaultEventTypes => EventListenerToLog.ContentionTypes;
+
+    protected override Action<ContentionEvent> OnEvent { get; }
+
+    private readonly Dictionary<Guid, DateTime> _contentionStarts = new ();
+
+    public ContentionEventsHandler(Action<ContentionEvent> onEvent, HashSet<EventType> eventTypes = null, long minimumDurationInMs = 0)
+    {
+        Update(eventTypes, minimumDurationInMs);
+        OnEvent = onEvent;
+    }
+
+    public override bool HandleEvent(EventWrittenEventArgs eventData)
+    {
+        switch (eventData.EventName)
+        {
+            case "ContentionStart":
```

Comment:
maybe put those in constants, all of them

Reply:
done